src/signer.rs

```diff
@@ -1,50 +1,127 @@
 use core::marker::PhantomData;
 
-use hashbrown::HashMap;
+use codec::{Decode, Encode, MaxEncodedLen};
+use frame_support::{pallet_prelude::RuntimeDebug, traits::Get, BoundedBTreeMap};
+use frame_system::Config as SysConfig;
 use move_core_types::account_address::AccountAddress;
+use scale_info::TypeInfo;
 
-use crate::{balance::BalanceOf, Config, Error, Pallet};
+use crate::{
+    balance::{BalanceAdapter, BalanceOf},
+    Config, Error, Pallet,
+};
+
+/// Generic definition of the hash of an extrinsic call.
+pub type CallHash = [u8; 32];
 
 /// A simple signature.
-#[derive(PartialEq, Eq)]
-enum Signature {
+#[derive(Clone, Eq, PartialEq, Encode, Decode, Default, RuntimeDebug, TypeInfo, MaxEncodedLen)]
+pub enum Signature {
     /// Signer has approved the script with the signature.
     Approved,
     /// Signer's approval still missing.
+    #[default]
     Missing,
 }
 
+#[derive(Clone, Eq, PartialEq)]
+pub enum MultisignError {
```

Comment:
```suggestion
pub enum MultiSignError {
```

Reply:
or MultiSig if we decide to use that naming everywhere

Reply:
Choose `MultisigError` since it is quite related to `Multisig`